### PR TITLE
[ADF-5378] Rotate image: auto increment version on submit

### PR DESCRIPTION
**Pull request by ursedaniel**
**Please check if the PR fulfills these requirements**

> - [ ] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

Draft PR in order for team members to test/demo.

**What is the new behaviour?**

https://github.com/Alfresco/alfresco-ng2-components/pull/6958
https://alfresco.atlassian.net/browse/ACS-1374

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Review by eromano — CHANGES_REQUESTED** [review]
unit test missing

**Review by DenysVuika — APPROVED** [review]

File: src/app/store/effects/upload.effects.ts [hunk 1]
```diff
@@ -132,6 +143,24 @@ export class UploadEffects {
       });
   }
 
+  uploadNewImage(file: File): void {
+    this.contentService.getNodeInfo().subscribe((node) => {
+      if (node && node.id) {
```
Comment (mauriziovitale) [inline src/app/store/effects/upload.effects.ts:148]: can we use node?.id

Comment (ursedaniel) [reply src/app/store/effects/upload.effects.ts]: changed, thanks for suggestion

File: package.json [hunk 2]
```diff
@@ -47,6 +47,7 @@
     "@ngrx/store": "^9.2.0",
     "@ngrx/store-devtools": "^9.2.0",
     "@ngx-translate/core": "^13.0.0",
+    "cropperjs": "1.5.11",
```
Comment (popovicsandras) [inline package.json:50]: Have you created a PR for ADW as well? Merging this PR might make ADW failed, if it doesn't have the same packages.

**Review by popovicsandras — APPROVED** [review]

**Review by eromano — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (ursedaniel) [reply package.json:50]: gonna do that now, thanks.

File: src/app/store/effects/upload.effects.ts [hunk 3]
```diff
@@ -132,6 +143,24 @@ export class UploadEffects {
       });
   }
 
+  uploadNewImage(file: File): void {
+    this.contentService.getNodeInfo().subscribe((node) => {
```
Comment (mauriziovitale) [inline src/app/store/effects/upload.effects.ts:147]: can we use the await? and try catch for the error

File: src/app/store/effects/upload.effects.spec.ts [hunk 4]
```diff
@@ -210,4 +210,15 @@ describe('UploadEffects', () => {
       expect(contentManagementService.versionUpdateDialog).toHaveBeenCalledWith(fakeEvent.detail.data.node.entry, fakeEvent.detail.files[0].file);
     });
   });
+
+  describe('image versioning', () => {
+    it('should trigger upload file version from viewer', () => {
```
Comment (mauriziovitale) [inline src/app/store/effects/upload.effects.spec.ts:215]: please make sure we test the error for this.contentService.getNodeInfo()

**Review by mauriziovitale — CHANGES_REQUESTED** [review]
please check the comments

File: src/app/store/effects/upload.effects.spec.ts (hunk 4 above)
Comment (ursedaniel) [reply src/app/store/effects/upload.effects.spec.ts:264]: tested

File: src/app/store/effects/upload.effects.ts (hunk 3 above)
Comment (ursedaniel) [reply src/app/store/effects/upload.effects.ts]: changed with better rxjs approach

**Review by DenysVuika — APPROVED** [review]

**Pull request merged**